### PR TITLE
refactor: resolve dependence problem

### PR DESCRIPTION
**Pull request by guanbinrui**
## Description

We have some cross-chain shared UI need to depend on `@masknet/plugin-infra`. It's straight to put those components in the `@masknet/shared`. But it runs into circular dependence.

```bash
@masknet/shared <- @masknet/plugin-infra <- @masknet/shared
```

This pull request moves those utilities required by `@masknet/plugin-infra` in `@masknet/shared` to `@masknet/shared-base` to break the circular dependence.

Closes # (NO_ISSUE)

## Type of change

- [ ] Documentation
- [x] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

**Review by UncleBill — DISMISSED** [review]

File: packages/shared-base/package.json [hunk 1]
```diff
@@ -21,10 +21,12 @@
     "immer": "^9.0.12",
     "pvtsutils": "^1.2.2",
     "react": "../empty",
+    "react-use": "^17.3.2",
```
Comment (Jack-Works) [inline packages/shared-base/package.json:24]: hello, please don't do this. shared-base should never export a react class. it's pure and can be run in any environment.

**Review by Jack-Works — CHANGES_REQUESTED** [review]

File: packages/shared-base/package.json (hunk 1 above)
Comment (Jack-Works) [reply packages/shared-base/package.json:24]: or use any react feature. as you can see, I explicitly resolves `react` to an empty package within this `shared-base` package.

File: .i18n-codegen.json [hunk 2]
```diff
@@ -38,6 +38,18 @@
                 "sourceMap": "inline"
             }
         },
+        {
+            "input": "./packages/shared-base/src/UI/locales/en-US.json",
```
Comment (Jack-Works) [inline .i18n-codegen.json:42]: and it should not have i18n features.

**Comment by Jack-Works** [comment]
per changes you made and the constraints I have, I suggest closing this PR and reconsidering another solution.

File: packages/shared-base/package.json (hunk 1 above)
Comment (guanbinrui) [reply packages/shared-base/package.json:24]: I have to add another package for only shared-base UI.

**Comment by Jack-Works** [comment]
For `CrashUI` and `ErrorBoundary`, I suggest you move it to the `theme` package, for those hooks, you may add a new package to store those hooks, or just leave them in the `shared`.

I'm closing this one because start from scratch will be easier than revert the work on this PR.

**Comment by guanbinrui** [comment]
> For `CrashUI` and `ErrorBoundary`, I suggest you move it to the `theme` package, for those hooks, you may add a new package to store those hooks, or just leave them in the `shared`.

I quite package panic now.

**Pull request closed without merging**